docs/options api.md

```diff
@@ -28,47 +28,54 @@ Choice, and defining `alias_true = option_full`.
 and is reserved by AP. You can set this as your default value, but you cannot define your own `option_random`.
 
 As an example, suppose we want an option that lets the user start their game with a sword in their inventory. Let's
-create our option class (with a docstring), give it a `display_name`, and add it to a dictionary that keeps track of our
-options:
+create our option class (with a docstring), give it a `display_name`, and add it to our game's options dataclass:
 
 ```python
 # Options.py
+from dataclasses import dataclass
+
+from Options import Toggle, PerGameCommonOptions
+
+
 class StartingSword(Toggle):
     """Adds a sword to your starting inventory."""
     display_name = "Start With Sword"
 
 
-example_options = {
-    "starting_sword": StartingSword
-}
+@dataclass
+class ExampleGameOptions(PerGameCommonOptions):
+    starting_sword: StartingSword
 ```
 
 This will create a `Toggle` option, internally called `starting_sword`. To then submit this to the multiworld, we add it
 to our world's `__init__.py`:
 
 ```python
 from worlds.AutoWorld import World
-from .Options import options
+from .Options import ExampleGameOptions
 
 
 class ExampleWorld(World):
-    option_definitions = options
+    # this gives the generator all the definitions for our options
+    options_dataclass = ExampleGameOptions
+    # this gives us typing hints for all the options we defined
+    options: ExampleGameOptions
 ```
 
 ### Option Checking
 Options are parsed by `Generate.py` before the worlds are created, and then the option classes are created shortly after
 world instantiation. These are created as attributes on the MultiWorld and can be accessed with
-`self.multiworld.my_option_name[self.player]`. This is the option class, which supports direct comparison methods to
+`self.o.my_option_name`. This is the option class, which supports direct comparison methods to
```

Comment:
```suggestion
`self.options.my_option_name`. This is an instance of the option class, which supports direct comparison methods to
```
note and double-check the 2 changes here:

`self.o` or `self.options`?

Normally I wouldn't be so pedantic between "instance" and "class", but in this case, we do a lot with the class objects that aren't instances of the class, so it's important to note the difference.